client/src/components/SignUp.js

```diff
@@ -1,13 +1,177 @@
 import React, { Component, Fragment } from 'react';
+import axios from 'axios';
+
+import * as S from './Login.style';
+
+class signUp extends Component {
+  state = {
+    name: '',
+    email: '',
+    password: '',
+    nameError: '',
+    emailError: '',
+    passwordError: '',
+    isErrorName: false,
+    isErrorEmail: '',
+    isErrorPassword: false,
+  };
+
+  componentDidMount() {
+    axios.get('/checkCookie').then(({ data: { cookie } }) => {
+      if (cookie) {
+        const { history } = this.props;
+        history.push('/account/login');
+      }
+    });
+  }
+
+  validate = () => {
+    let isError = false;
+    this.setState({
+      isErrorName: false,
+      isErrorEmail: false,
+      isErrorPassword: false,
+    });
+    const name = this.state;
+    const characters = [
+      '+',
+      '/',
+      '*',
+      '$',
+      '#',
+      '@',
+      '!',
+      '^',
+      '&',
+      '(',
+      ')',
+      '?',
+      '>',
+      '<',
+      '.',
+    ];
+    const errors = {
+      nameError: '',
+      emailError: '',
+      passwordError: '',
+      isErrorName: false,
+      isErrorEmail: false,
+      isErrorPassword: false,
+    };
+    for (let i = 0; i < characters.length; i++) {
```

Comment:
should create a function comparing password to password 2 to make sure that they are the same